exp/api/v1beta1/azuremachinepoolmachine_types.go

```diff
@@ -67,7 +67,7 @@ type (
 		// can be added as events to the MachinePool object and/or logged in the
 		// controller's output.
 		// +optional
-		FailureReason *errors.MachineStatusError `json:"failureReason,omitempty"`
+		FailureReason *errors.MachinePoolStatusFailure `json:"failureReason,omitempty"`
```

Comment:
@Jont828 Do updates to `MachinePoolMachine` type also qualify as right change?

Reply:
I think we can leave it as `MachineStatusError`. My understanding is that this would be set with the error related to this specific AzureMachinePoolMachine and not the AzureMachinePool.